master/internal/db/bun_filters.go

```diff
@@ -140,3 +141,180 @@ func ApplyTimestampFieldFilter[T string | schema.Ident](
 
 	return q, nil
 }
+
+// ValidateInt32FieldFilterComparison validates the min and max values in the range.
+func ValidateInt32FieldFilterComparison(
+	filter *commonv1.Int32FieldFilter,
+) error {
+	var minValue, maxValue int32
+	if filter == nil {
+		return nil
+	}
+	if filter.Gt == nil && filter.Gte == nil {
+		return nil
+	}
+	if filter.Lt == nil && filter.Lte == nil {
+		return nil
+	}
+	if filter.Lt != nil && filter.Lte != nil { //nolint: gocritic
+		maxValue = mathx.Max(*filter.Lt, *filter.Lte)
+	} else if filter.Lt != nil {
+		maxValue = *filter.Lt
+	} else {
+		maxValue = *filter.Lte
+	}
+	if filter.Gt != nil && filter.Gte != nil { //nolint: gocritic
+		minValue = mathx.Min(*filter.Gt, *filter.Gte)
```

Comment:
issue: this logic will allow `gt=4&lt=5` even though this will never return anything. 
in my opinion this is low priority thing which can be a TODO. mutually exclusive filtering which returns nothing without an explicit error is an acceptable behavior.